### PR TITLE
Bar component dom

### PR DESCRIPTION
**Pull request by dommiller88**
cmon boys

### CODE REVIEW

File: examples/bar/main.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    let children: Vec<Box<dyn ComponentTraits>> = Vec::new();
+    fn nothing() {}
+    let mut bar: BarContainer = BarContainer::new(
+        nothing,
+        true,
+        375,
+        667,
+        0,
+        0,
+        children,
```
Comment (nicballesteros) [inline examples/bar/main.rs]: `Option<Vec...>`
`Some(children)`
`None`,
`if children == None` Initialize the vector to none.

File: src/framework/components/bar.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    }
+                }
+                else {
+                    println!("Insufficient vertical space in container.");
+                }
+            }
+        }
+        
+    }
+
```
Comment (nicballesteros) [inline src/framework/components/bar.rs]: remove extra spaces

File: src/framework/components/bar.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    
+}
+impl Draw for BarContainer{
+    fn draw(&self, canvas: &mut Canvas) {
+        if (self.visible) {
+            for child in self.children.iter() {
+                child.draw(canvas);
+            }
+        }
+
```
Comment (nicballesteros) [inline src/framework/components/bar.rs]: space

File: src/orchestrator/redraw.rs [hunk 4]
```diff
@@ -1,7 +1,9 @@
 use skia_safe::{ Color, Canvas, Rect, Color4f };
 use skia_safe::paint::{ Paint };
 
-use crate::components::BoxComponent;
+use crate::components::*;
+// use crate::components::Draw;
```
Comment (nicballesteros) [inline src/orchestrator/redraw.rs]: remove comments

**Review by nicballesteros — DISMISSED** [review]

**Review by Lirons01 — DISMISSED** [review]
lookin good

**Review by nicballesteros — APPROVED** [review]

**Pull request merged**